src/coreclr/src/vm/codeversion.cpp

```diff
@@ -1670,6 +1671,14 @@ PCODE CodeVersionManager::PublishVersionableCodeIfNecessary(
         {
             PrepareCodeConfigBuffer configBuffer(activeVersion);
             PrepareCodeConfig *config = configBuffer.GetConfig();
+
+#ifdef FEATURE_TIERED_COMPILATION
+            // If the caller is in preemptive mode, then
+            // Tiered Compilation isn't supported. (e.g. NativeCallable)
```

Comment:
I believe that we can come here with `CallerGCMode_Unknown`, e.g. if somebody calls `RuntimeHelpers.PrepareMethod` on the method. Maybe add test for it?

If we come here with unknown, we need to lookup the attribute the expensive way.

Reply:
Oh wow. I just realized what this means. Okay this definitely warrants a test. Thanks.